enterprise/signals.py

```diff
@@ -99,17 +99,24 @@ def _complete_user_enrollment():
         pending_ecu.delete()
 
 
+@receiver(pre_save, sender=EnterpriseCustomerBrandingConfiguration)
+def skip_saving_logo_file(sender, instance, **kwargs):     # pylint: disable=unused-argument
+    """
+    To avoid saving the logo image at an incorrect path, skip saving it.
+    """
+    if not instance.id and not hasattr(instance, _UNSAVED_FILEFIELD):
+        setattr(instance, _UNSAVED_FILEFIELD, instance.logo)
+        instance.logo = None
+
+
 @receiver(post_save, sender=EnterpriseCustomerBrandingConfiguration)
-def update_logo_name(sender, instance, **kwargs):     # pylint: disable=unused-argument
+def save_logo_file(sender, instance, **kwargs):            # pylint: disable=unused-argument
     """
-    Update logo name and path to replace None with instance id.
+    Now that the object is instantiated and instance.id exists, save the image at correct path and re-save the model.
     """
-    filename = instance.logo.name
-    if kwargs['created'] and 'None' in filename:
-        extension = os.path.splitext(filename)[1].lower()
-        instance_file = instance.logo.file
-        new_name = 'enterprise/branding/' + str(instance.id) + '/' + str(instance.id) + '_logo' + extension
-        instance.logo.save(new_name, instance_file)
+    if kwargs['created'] and hasattr(instance, _UNSAVED_FILEFIELD):
+        instance.logo = getattr(instance, _UNSAVED_FILEFIELD)
+        instance.__dict__.pop(_UNSAVED_FILEFIELD)
```

Comment:
can we use `delattr` builtin function here?

Reply:
Good suggestion. I have incorporated that.